community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -1002,13 +1007,10 @@ private StatementOperationParts buildStatementOperations(
         parts = parts.override( null, null, null, lockingContext, lockingContext, lockingContext, lockingContext,
                 lockingContext, null, null, null );
         // + Guard
-        if ( guard != null )
-        {
-            GuardingStatementOperations guardingOperations = new GuardingStatementOperations(
-                    parts.entityWriteOperations(), parts.entityReadOperations(), guard );
-            parts = parts.override( null, null, guardingOperations, guardingOperations, null, null, null, null,
-                    null, null, null );
-        }
+        GuardingStatementOperations guardingOperations = new GuardingStatementOperations(
```

Comment:
This will cause the guard layer to always be there, causing more delegation calls for all users. I think we should only have this layer on if there's a timeout configured.